client/python/datajunction/client.py

```diff
@@ -215,7 +216,281 @@ def data(  # pylint: disable=too-many-arguments,too-many-locals
             )
 
 
-class DJWriter(DJReader):
+class DJBuilder(DJClient):
     """
-    Client class to manage all your DJ dag: nodes, attributes, namespaces, tags.
+    Client class for DJ node manipulation.
     """
+
+    #
+    # Catalogs and Engines
+    #
+    def list_catalogs(self):
+        """
+        Gets all catalogs.
+        """
+        response = self._session.get("/catalogs/", timeout=self._timeout)
+        return response.json()
+
+    def list_engines(self):
+        """
+        Gets all engines.
+        """
+        response = self._session.get("/engines/", timeout=self._timeout)
+        return response.json()
+
+    #
+    # Namespace
+    #
+    # TODO: add delete namespace (with warnings) # pylint: disable=fixme
+    def namespace(self, namespace: str) -> "Namespace":
+        """
+        Returns the specified node namespace.
+        """
+        try:
+            return self.create_namespace(namespace=namespace)
+        except DJNamespaceAlreadyExists:
+            pass
+        return Namespace(namespace=namespace, dj_client=self)
+
+    def create_namespace(self, namespace: str) -> "Namespace":
```

Comment:
The create prefix implies that the this method would actually create it, but it won't since the user still has to call `.save()`. That's why we went with the new prefix but maybe we were overthinking. @shangyian what do you think?

Reply:
Yes, I changed that pattern from `new + save` to `create`.